packages/react-components/react-combobox/src/components/Combobox/Combobox.test.tsx

```diff
@@ -23,10 +24,386 @@ describe('Combobox', () => {
     },
   });
 
-  // TODO add more tests here, and create visual regression tests in /apps/vr-tests
-
   it('renders a default state', () => {
-    const result = render(<Combobox>Default Combobox</Combobox>);
+    const result = render(
+      <Combobox>
+        <Option>Red</Option>
+        <Option>Green</Option>
+        <Option>Blue</Option>
+      </Combobox>,
+    );
+    expect(result.container).toMatchSnapshot();
+  });
+
+  it('renders an open listbox', () => {
+    const result = render(
+      <Combobox open inline>
+        <Option>Red</Option>
+        <Option>Green</Option>
+        <Option>Blue</Option>
+      </Combobox>,
+    );
     expect(result.container).toMatchSnapshot();
   });
+
+  it('renders the popup under document.body by default', () => {
+    const { container } = render(
+      <Combobox open>
+        <Option>Red</Option>
+        <Option>Green</Option>
+        <Option>Blue</Option>
+      </Combobox>,
+    );
+    expect(container.querySelector('[role=listbox]')).toBeNull();
+    expect(document.body.querySelector('[role=listbox]')).not.toBeNull();
+  });
+
+  it('renders the popup inline when specified', () => {
+    const { container } = render(
+      <Combobox open inline>
+        <Option>Red</Option>
+        <Option>Green</Option>
+        <Option>Blue</Option>
+      </Combobox>,
+    );
+    expect(container.querySelector('[role=listbox]')).not.toBeNull();
+  });
+
+  /* open/close tests */
+  it('opens the popup on click', () => {
+    const { getByRole } = render(
+      <Combobox>
+        <Option>Red</Option>
+        <Option>Green</Option>
+        <Option>Blue</Option>
+      </Combobox>,
+    );
+
+    fireEvent.click(getByRole('combobox'));
+
+    expect(getByRole('listbox')).not.toBeNull();
+    expect(getByRole('combobox').getAttribute('aria-expanded')).toEqual('true');
+  });
+
+  it('closes the popup on click with defaultOpen', () => {
+    const { getByRole, queryByRole } = render(
+      <Combobox defaultOpen>
+        <Option>Red</Option>
+        <Option>Green</Option>
+        <Option>Blue</Option>
+      </Combobox>,
+    );
+
+    const combobox = getByRole('combobox');
+
+    expect(getByRole('listbox')).not.toBeNull();
+    expect(combobox.getAttribute('aria-expanded')).toEqual('true');
+
+    fireEvent.click(combobox);
+
+    expect(queryByRole('listbox')).toBeNull();
+    expect(combobox.getAttribute('aria-expanded')).toEqual('false');
+  });
+
+  it('does not open the combobox on click with controlled open', () => {
```

Comment:
This should read `close` instead of `open` right?

```suggestion
  it('does not close the combobox on click with controlled open', () => {
```

Reply:
lol yup